### PR TITLE
#1 Add template suggestions for view modes and standardise body classes

### PR DESCRIPTION
**Pull request by justafish**

### CODE REVIEW

File: template.php [hunk 1]
```diff
@@ -50,11 +50,64 @@ function windup_process_html(&$variables, $hook) {
...
+  }
+  if ($variables['sticky']) {
+    $variables['classes_array'][] = 'status--sticky';
+  }
+  if (!$variables['status']) {
+    $variables['classes_array'][] = 'status--unpublished';
+  }
+  if (isset($variables['preview'])) {
+    $variables['classes_array'][] = 'status--preview';
+  }
```
Comment (krisbulman) [inline template.php]: Lets stick with the [D8 node status classes](https://api.drupal.org/api/drupal/core!modules!node!templates!node.html.twig/8)

```
node--promoted
node--sticky
node--unpublished
...etc
```

Comment (justafish) [inline template.php]: I'd rather not do this actually, because this isn't specific to nodes, it's used on all entities.

File: template.php [hunk 2]
```diff
@@ -50,11 +50,64 @@ function windup_process_html(&$variables, $hook) {
...
+
+  // Add templates for node view modes.
+  $variables['theme_hook_suggestions'][] = 'node__' . $variables['type'] . '__' . $variables['view_mode'];
+  if (!empty($variables['id'])) {
+    $variables['theme_hook_suggestions'][] = 'node__' . $variables['type'] . '__' . $variables['view_mode'] . '__' . $variables['id'];
+  }
+
+  // Add classes for the view mode.
+  $variables['classes_array'][] = drupal_html_class('node-' . $variables['type'] . '-' . $variables['view_mode']);
+  $variables['classes_array'][] = drupal_html_class('view-mode--' . $variables['view_mode']);
```
Comment (krisbulman) [inline template.php]: Lets stick with the [D8 node classes](https://api.drupal.org/api/drupal/core!modules!node!templates!node.html.twig/8)

```
node--type-[type]-[view_mode]
node--view-mode-[view_mode]
```

File: template.php [hunk 3]
```diff
@@ -50,11 +50,64 @@ function windup_process_html(&$variables, $hook) {
...
   $variables['unpublished'] = (!$variables['status']) ? TRUE : FALSE;
+
+  // Add templates for node view modes.
+  $variables['theme_hook_suggestions'][] = 'node__' . $variables['type'] . '__' . $variables['view_mode'];
+  if (!empty($variables['id'])) {
+    $variables['theme_hook_suggestions'][] = 'node__' . $variables['type'] . '__' . $variables['view_mode'] . '__' . $variables['id'];
+  }
+
+  // Add classes for the view mode.
+  $variables['classes_array'][] = drupal_html_class('node-' . $variables['type'] . '-' . $variables['view_mode']);
```
Comment (krisbulman) [inline template.php]: if we're adding ```node-['type']-['view_mode']``` lets at least use ```node--```

File: template.php (hunk 1 above)
Comment (krisbulman) [inline template.php]: all good with me, thanks for reminding me of that fact.

**Comment by davereid** [comment]
I've generally recommended that any site just have the <a href="https://www.drupal.org/project/entity_view_mode</a>entity_view_mode</a> module enabled since it makes these view mode template suggestions available to *all* entity types. Is there a reason we should be encouraging the starter theme to be also adding this every time, but only for nodes?

**Comment by justafish** [comment]
@davereid While I would agree, I don't think we want to rely on someone having a particular module installed.

I've added a commit that will normalise template suggestions and classes across all entity types. See the README changes for details.

/cc @krisbulman @carwin

**Pull request merged**